base/abstractarraymath.jl

```diff
@@ -91,6 +91,10 @@ imag{T<:Real}(x::AbstractArray{T}) = zero(x)
 
 # index A[:,:,...,i,:,:,...] where "i" is in dimension "d"
 
+_slice_inds(new_inds::Tuple, inds::Tuple, d, i) =
+    _slice_inds((new_inds..., length(new_inds) + 1 == d ? i : first(inds)), tail(inds), d, i)
+_slice_inds(new_inds::Tuple, inds::Tuple{}, d, i) = new_inds
```

Comment:
Could also be
```julia
setindex(x::Tuple, n, v) = _setindex((), x, n, v)
_setindex(y::Tuple, x::Tuple, n, v) = _setindex((y..., length(y) + 1 == n ? v : first(x)), tail(x), n, v)
_setindex(y::Tuple, x::Tuple{}, n, v) = y
```
and live in tuple.jl. Or do we have something like this somewhere already?